packages/astro/test/test-utils.js

```diff
@@ -243,6 +243,20 @@ export async function loadFixture(inlineConfig) {
 	};
 }
 
+/**
+ * 
+ * @param {string} text
+ * @returns {{ componentUrl: string, componentExport: string, rendererUrl: string, beforeHydrationUrl?: string, opts: Record<string, unknown>, props: Record<string, unknown> }}
+ */
+export function getIslandDataFromScript(text) {
+	const Astro = { assign: (_, props) => props };
+	const document = {};
+	const args = { Astro, document }
+
+	// Yes, I know, `eval` is bad! But our data is embedded as JS on purpose.
+	return new Function(Object.keys(args), `return ${text}`)(...Object.values(args));
+}
```

Comment:
Calling this out! New test util that extracts data from the `astro-island > script` text. It's using `new Function` because the data is now embedded as JavaScript rather than JSON, so we need to `eval` it. 

However, we don't serialize functions so this content is all safe, and the function is scoped to receive the local `Astro` and `document` args defined here.